### ISSUE TITLE
Virtual Service Delay bug

### ISSUE THREAD

**Issue by chenhy97** [snowdrop/istio-java-api#103]
Istio:1.6.0
kubernetes-client:4.9.0
istio-java-api:1.6.3-beta1
When creating VirtualService with Delay Fault which is defined with "percentage", I encountered a bug:
```
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Unknown field 'percentage' (through reference chain: me.snowdrop.istio.api.networking.v1alpha3.VirtualService["spec"]->me.snowdrop.istio.api.networking.v1alpha3.VirtualServiceSpec["http"]->java.util.ArrayList[0]->me.snowdrop.istio.api.networking.v1alpha3.HTTPRoute["fault"]->me.snowdrop.istio.api.networking.v1alpha3.HTTPFaultInjection["delay"])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:394)
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:353)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.wrapAndThrow(BeanDeserializerBase.java:1714)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:290)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:286)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:245)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:27)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4202)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3250)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:246)
```
However, Virtual Service with Abort Fault is normal.

What is more, the api version of the created VIrtual Service is "networking.istio.io/v1beta1" while I use "networking.istio.io/v1alpha3" in my code.

**Comment by metacosm** [comment]
Do you have a code sample / YAML file that I could look at?

**Comment by chenhy97** [comment]
@metacosm 
The code:
```
Duration fixedDelay = new Duration(0, (long)dur_or_code);
FixedDelayHttpDelayType fDelayHttpDelayType = new FixedDelayHttpDelayType(fixedDelay);
Delay delay = new Delay();
delay.setHttpDelayType(fDelayHttpDelayType);
Percent percentage = new Percent(perc);
delay.setPercentage(perc);
fault.setDelay(delay);
virtualservice = new VirtualServiceBuilder().withApiVersion("networking.istio.io/v1alpha3")
                        .withNewMetadata().withName(vs_name).withNamespace(ns).endMetadata()
                        .withNewSpec().withHosts(svc_name)
                        .addNewHttp()
                        .withMatch(hMatchRequest)
                        .withRoute(hRouteDestination)
                        .withFault(fault)
                        .endHttp().endSpec()
                        .build();
```
Yaml:
```
apiVersion: networking.istio.io/v1beta1
kind: VirtualService
metadata:
  creationTimestamp: "2020-07-21T13:44:52Z"
  generation: 1
  name: test-vs
  namespace: ts
  resourceVersion: "6119588"
  selfLink: /apis/networking.istio.io/v1beta1/namespaces/ts/virtualservices/test-vs
  uid: c0393d6d-c8a9-4189-a576-e3d86ff5f960
spec:
  hosts:
  - search-planner
  http:
  - fault:
      delay:
        fixedDelay: 6s
        percentage:
          value: 90.0
    match:
    - uri:
        exact: /search
    route:
    - destination:
        host: search-planner
        port:
          number: 8080
      headers: {}
      weight: 100
```

**Comment by metacosm** [comment]
@chenhy97 thank you, I am going to take a look

**Comment by metacosm** [comment]
> What is more, the api version of the created VIrtual Service is "networking.istio.io/v1beta1" while I use "networking.istio.io/v1alpha3" in my code.

Regarding that part, are you sure that you import the `v1alpha3` version of the `VirtualServiceBuilder` class?

**Comment by metacosm** [comment]
Please give a try to newly released 1.6.5-Beta3.
